pyvista/core/filters/data_set.py

```diff
@@ -1755,7 +1762,9 @@ def outline(self, generate_faces: bool = False, progress_bar: bool = False):
         _update_alg(alg, progress_bar, 'Producing an outline')
         return wrap(alg.GetOutputDataObject(0))
 
-    def outline_corners(self, factor=0.2, progress_bar: bool = False):
+    def outline_corners(  # type: ignore[misc]
+        self: _vtk.vtkDataSet | _vtk.vtkMultiBlockDataSet, factor=0.2, progress_bar: bool = False
```

Comment:
Adding a note here for completeness (not intended to block approval).

This is a tricky filter to type since the internal vtk filters allows vtkDataObject, not just vtkDataSet.
It would probably be better to use a new `ConcreteDataObjectType` type here (similar to `ConcreteDataSetType` added in https://github.com/pyvista/pyvista/pull/6854). Otherwise, if a `DataObject` is passed to this function a false positive type error will be generated (since `DataObject` does not inherit from `vtkDataObject`).

But, this would also (perhaps unfavorably) narrow the type to pyvista DataObjects only, even though in this case any vtkDataObject is allowed, resulting in false positive type errors for unwrapped vtk classes.

Perhaps it may be best to do `self: DataObject | vtkDataObject` and then wrap `self` in the filter. This general idea of typing `vtk | pyvista` inputs and wrapping `self` could apply to all filters. And it would help remove the use of `ConcreteDataSetType` or `ConcreteDataObjectType` as a type hint, which is good because this is more of an internal typing construct rather than something that users should have to think about.

These subtle typing changes can be made in a follow-up PR though. The main goal for this PR is to add initial annotations and get approval for the `type: ignore[misc]` idea generally.